javalib/src/main/scala/java/util/Properties.scala

```diff
@@ -81,6 +120,261 @@ class Properties(protected val defaults: Properties)
+          if (!prevValueContinue) {
+            setProperty(key, value)
+            isKeyParsed = false
+          }
+        } else if (prevValueContinue && valueContinues()) {
+          val value = parseValue()
+          prevValueContinue = valueContinues()
+        } else {
+          val value = parseValue()
+          setProperty(key, value)
+          isKeyParsed = false
+          prevValueContinue = false
+        }
+      }
+    }
+  }
+
+  private def writeComments(writer: Writer, comments: String,
+      toHex: Boolean): Unit = {
+    writer.write('#')
+    val chars = comments.toCharArray
+    var index = 0
+    while (index < chars.length) {
+      val ch = chars(index)
+      if (ch <= 0xff) {
+        if (ch == '\r' || ch == '\n') {
+          def isCrlf =
+            ch == '\r' && index + 1 < chars.length && chars(index + 1) == '\n'
+
+          if (isCrlf) {
+            index += 1
+          }
+          writer.write(System.lineSeparator)
+
+          def noExplicitComment = {
+            index + 1 < chars.length &&
+            (chars(index + 1) != '#' && chars(index + 1) != '!')
+          }
+
+          if (noExplicitComment) {
+            writer.write('#')
+          }
+        } else {
+          writer.write(ch)
+        }
+      } else {
+        if (toHex) {
+          writer.write(unicodeToHexaDecimal(ch))
+        } else {
+          writer.write(ch)
+        }
+      }
+      index += 1
+    }
+    writer.write(System.lineSeparator)
+  }
+
+  private def encodeString(string: String, isKey: Boolean,
+      toHex: Boolean): String = {
+    val buffer = new jl.StringBuilder(200)
+    var index = 0
+    val length = string.length
+    // leading element (value) spaces are escaped
+    if (!isKey) {
+      while (index < length && string.charAt(index) == ' ') {
+        buffer.append("\\ ")
+        index += 1
+      }
+    }
+
+    while (index < length) {
+      val ch = string.charAt(index)
+      (ch: @switch) match {
+        case '\t' =>
+          buffer.append("\\t")
+        case '\n' =>
+          buffer.append("\\n")
+        case '\f' =>
+          buffer.append("\\f")
+        case '\r' =>
+          buffer.append("\\r")
+        case '\\' | '#' | '!' | '=' | ':' =>
+          buffer.append('\\')
+          buffer.append(ch)
+        case ' ' if isKey =>
+          buffer.append("\\ ")
+        case _ =>
+          if (toHex && (ch < 0x20 || ch > 0x7e)) {
+            buffer.append(unicodeToHexaDecimal(ch))
+          } else {
+            buffer.append(ch)
+          }
+      }
+      index += 1
+    }
+    buffer.toString()
+  }
+
+  private def unicodeToHexaDecimal(ch: Int): Array[Char] = {
+    def hexChar(x: Int): Char =
+      if (x > 9) (x - 10 + 'A').toChar
+      else (x + '0').toChar
+
+    Array('\\',
+          'u',
+          hexChar((ch >>> 12) & 15),
+          hexChar((ch >>> 8) & 15),
+          hexChar((ch >>> 4) & 15),
+          hexChar(ch & 15))
```

Comment:
Somehow I overlooked this the previous time.
```suggestion
    Array(
        '\\',
        'u',
        hexChar((ch >>> 12) & 15),
        hexChar((ch >>> 8) & 15),
        hexChar((ch >>> 4) & 15),
        hexChar(ch & 15)
    )
```

Reply:
Do you have an easy way to do the the squash? I have only done it like once and I am having problems.

Reply:
`git rebase -i`? https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

Reply:
I thought maybe there was something easier.